backends/tfhe-cuda-backend/cuda/src/zk/expand.cuh

```diff
@@ -0,0 +1,73 @@
+#ifndef EXPAND_CUH
+#define EXPAND_CUH
+
+#include "device.h"
+#include "polynomial/functions.cuh"
+#include "polynomial/polynomial_math.cuh"
+#include "zk/zk.h"
+#include <cstdint>
+
+#include "utils/helper.cuh"
+
+// Expand a LweCompactCiphertextList into a LweCiphertextList
+// - Each x-block computes one output ciphertext
+template <typename Torus, class params>
+__global__ void lwe_expand(Torus const *lwe_compact_array_in,
+                           Torus *lwe_array_out,
+                           const uint32_t *compact_list_length_per_thread,
+                           const uint32_t *lwe_compact_input_indexes) {
+  const auto lwe_output_id = blockIdx.x;
+  const auto lwe_dimension = params::degree;
+
+  const auto compact_list_length =
+      compact_list_length_per_thread[lwe_output_id];
+  const auto body_id = lwe_output_id % compact_list_length;
```

Comment:
Does this really work? If previous compact lists don't have the same length as this one, won't it be an issue? :thinking:

Reply:
Not sure I'm following. We give to the kernel the number of LWEs stored in the compact list associated to each `blockIdx.x` (which relates to an output LWE) , and also the index that that compact list will be found in the input array. The kernel won't care about the size of other compact lists if it knows exactly where to access things.